### PR TITLE
Remove ObservableCollection from HotspotsStore

### PR DESCRIPTION
**Pull request by rita-g-sonarsource**

### CODE REVIEW

File: src/IssueViz.Security/Hotspots/HotspotsStore.cs [hunk 1]
```diff
@@ -65,14 +59,22 @@ public IAnalysisIssueVisualization GetOrAdd(IAnalysisIssueVisualization hotspotV
                 return existingHotspot;
             }
 
+            var oldIssues = hotspots.ToArray();
+
             hotspots.Add(hotspotViz);
 
+            NotifyIssuesChanged(oldIssues);
```
Comment (duncanp-sonar) [inline src/IssueViz.Security/Hotspots/HotspotsStore.cs:66]: The only change to the collection is the addition of a new issue and it isn't part of the notification. Is this correct?

File: src/IssueViz.Security/Hotspots/HotspotsStore.cs [hunk 2]
```diff
@@ -65,14 +59,22 @@ public IAnalysisIssueVisualization GetOrAdd(IAnalysisIssueVisualization hotspotV
...
             return hotspotViz;
         }
 
         public void Remove(IAnalysisIssueVisualization hotspotViz)
         {
+            var oldIssues = hotspots.ToArray();
+
             hotspots.Remove(hotspotViz);
+
+            NotifyIssuesChanged(oldIssues);
```
Comment (duncanp-sonar) [inline src/IssueViz.Security/Hotspots/HotspotsStore.cs:77]: This seems odd too - why are we saying "everything's changed" when only one issue has been removed?

File: src/IssueViz.Security/Hotspots/HotspotsStore.cs (hunk 1 above)
Comment (duncanp-sonar) [reply src/IssueViz.Security/Hotspots/HotspotsStore.cs:66]: Ah - I see. Seems a bit weird to return all of the old and new issue in the event arguments. I missed that before - I incorrectly interpreted it as "removed" and "added".

**Review by duncanp-sonar — APPROVED** [review]

File: src/IssueViz.Security/Hotspots/HotspotsStore.cs (hunk 1 above)
Comment (rita-g-sonarsource) [reply src/IssueViz.Security/Hotspots/HotspotsStore.cs:66]: Ah, but I do realize that there is a bug in the StoreAdapter: I need to notify only on what's changed, so a diff between old and new 🤦 I'll fix that in a separate PR.

**Pull request merged**